src/main/java/org/json/JSONArray.java

```diff
@@ -101,27 +113,38 @@ public JSONArray(JSONTokener x) throws JSONException {
                     this.myArrayList.add(JSONObject.NULL);
                 } else {
                     x.back();
-                    this.myArrayList.add(x.nextValue());
+                    if (jsonParserConfiguration.isStrictMode()) {
```

Comment:
This change is probably not needed. I think it might be sufficient to touch JSONTokener nextValue() and nextSimpleValue() to capture all all of the strict mode functionality. 

In nextValue(), if you pass in a JSONParserConfiguration, it can be passed to new JSONObject(), new JSONArray(), and nextSimpleValue()

In nextSimpleValue(), if you get to the return line, the value will have violated strict mode.

Of course, all of the callers of these methods will need a JSONParserConfiguration object to pass in.